packages/wasm/src/index.ts

```diff
@@ -0,0 +1,150 @@
+//@ts-ignore source map not found
+import {
+  BlsVariant,
+  EcdsaVariant,
+  InitOutput,
+  //@ts-ignore source map not found
+  getModule,
+  initSync,
+} from './pkg/wasm-internal';
+export type { BlsVariant, EcdsaVariant } from './pkg/wasm-internal';
+
+import {
+  blsCombine,
+  blsDecrypt,
+  blsEncrypt,
+  blsVerify,
+  ecdsaCombine,
+  ecdsaDeriveKey,
+  ecdsaVerify,
+  greet,
+  sevSnpGetVcekUrl,
+  sevSnpVerify,
+} from './pkg/wasm-internal';
+
+// loaded wasm module
+let wasmSdkInstance: InitOutput | undefined;
+
+export const initWASM = (): Promise<void> => {
+  return new Promise<void>((res, rej) => {
+    if (wasmSdkInstance) {
+      res();
+      return;
+    }
+    try {
+      // use wasm-bigen `init sync` for loading wasm modules.
+      // synchronously loads the module through `WebAssembly`
+      wasmSdkInstance = initSync(getModule());
+      res();
+    } catch (e) {
+      wasmSdkInstance = undefined;
+      rej(e);
+    }
+  });
+};
+
+/**
+ * async wrappers for wasm wrappers for future async implementations.
+ * Allows for lazy loadnig of WASM.
+ */
+
+export async function loadModuleAndBlsCombine(
```

Comment:
So we currently have to export the methods from the binding because `crypto` does not have async method signatures. I dont think we should change the interfaces in `crypto` at this time as it will force a major version upgrade. so this is why I add `loadModuleAnd` and kept the wrappers in the exports